heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -0,0 +1,419 @@
+                instanceId,
+                getComponentName(instanceId),
+                resource);
+        // Insert it into the map
+        instancePlanMap.put(instanceId, instancePlan);
+      }
+
+      containerCpu += (paddingPercentage * containerCpu) / 100;
+      containerRam += (paddingPercentage * containerRam) / 100;
+      containerDiskInBytes += (paddingPercentage * containerDiskInBytes) / 100;
+
+      PackingPlan.Resource resource =
+          new PackingPlan.Resource(Math.round(containerCpu), containerRam, containerDiskInBytes);
+
+      PackingPlan.ContainerPlan containerPlan =
+          new PackingPlan.ContainerPlan(containerId, instancePlanMap, resource);
+
+      containerPlanMap.put(containerId, containerPlan);
+      topologyRam += containerRam;
+      topologyCpu += Math.round(containerCpu);
+      topologyDisk += containerDiskInBytes;
+    }
+
+    // Take the heron internal container into account and the application master for YARN
+    // scheduler
+    topologyRam += instanceRamDefault;
+    topologyDisk += instanceDiskDefault;
+    topologyCpu += instanceCpuDefault;
+
+    PackingPlan.Resource resource = new PackingPlan.Resource(
+        topologyCpu, topologyRam, topologyDisk);
+
+    PackingPlan plan = new PackingPlan(topology.getId(), containerPlanMap, resource);
+    return plan;
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  /**
+   * Get the RAM requirements of all the components
+   *
+   * @return The list of components and their RAM requirements
+   */
+  protected ArrayList<RamRequirement> getRAMInstances() {
+    ArrayList<RamRequirement> ramRequirements = new ArrayList<>();
+    Map<String, Integer> parallelismMap = TopologyUtils.getComponentParallelism(topology);
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+
+    for (String component : parallelismMap.keySet()) {
+      if (ramMap.containsKey(component)) {
+        if (!isValidInstance(ramMap.get(component), instanceCpuDefault, instanceDiskDefault)) {
+          return null;
+        } else {
+          ramRequirements.add(new RamRequirement(component, ramMap.get(component)));
+        }
+      } else {
+        if (!isValidInstance(instanceRamDefault, instanceCpuDefault, instanceDiskDefault)) {
+          return null;
+        } else {
+          ramRequirements.add(new RamRequirement(component, instanceRamDefault));
+        }
+      }
+    }
+    return ramRequirements;
+  }
+
+  /**
+   * Get the instances' allocation based on the Resource Compliant Round Robin algorithm
+   *
+   * @return Map &lt; containerId, list of InstanceId belonging to this container &gt;
+   */
+  protected Map<String, List<String>> getResourceCompliantRRAllocation() {
+
+    Map<String, List<String>> allocation = new HashMap<>();
+    ArrayList<Container> containers = new ArrayList<>();
+    ArrayList<RamRequirement> ramRequirements = getRAMInstances();
+
+    if (ramRequirements == null) {
+      return null;
```

Comment:
@billonahill Actually they are both severe cases and they are caused my misconfiguration of the topology parameters. I added a log message in both cases.